models/contacts.js

```diff
@@ -1,8 +1,54 @@
 const Contact = require("./schemas/contact");
+const User = require("./schemas/users");
+const bcrypt = require("bcryptjs");
+const salt = bcrypt.genSaltSync(10);
+const jwt = require("jsonwebtoken");
+const secret = "secret word";
+const passportJWT = require("passport-jwt");
+const passport = require("passport");
+const ExtractJWT = passportJWT.ExtractJwt;
+const Strategy = passportJWT.Strategy;
+const params = {
+  secretOrKey: secret,
+  jwtFromRequest: ExtractJWT.fromAuthHeaderAsBearerToken(),
+};
+
+passport.use(
+  new Strategy(params, function (payload, done) {
+    User.find({ _id: payload.id })
+      .then(([user]) => {
+        if (!user) {
+          return done(new Error("User not found"));
+        }
+        return done(null, user);
+      })
+      .catch((err) => done(err));
+  })
+);
+const auth = (req, res, next) => {
+  passport.authenticate("jwt", { session: false }, (err, user) => {
+    if (!user || err) {
+      return res.status(401).json({
+        status: "error",
+        code: 401,
+        message: "Unauthorized",
+        data: "Unauthorized",
+      });
+    }
+    req.user = user;
```

Comment:
Never return password in response
![image](https://github.com/cmdspeed/nodejs/assets/142218823/084357d7-7ef3-45bf-9531-c68b959df6d5)

Reply:
also why do you have token in response for signup?